jaxb-ri/runtime/impl/src/main/java/com/sun/xml/bind/v2/runtime/reflect/opt/Injector.java

```diff
@@ -0,0 +1,363 @@
+        Injector injector = get(cl);
+        if (injector != null) {
+            return injector.find(className);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Gets or creates an {@link Injector} for the given class loader.
+     *
+     * @return null
+     *      if it fails.
+     */
+    private static Injector get(ClassLoader cl) {
+        Injector injector = null;
+        WeakReference<Injector> wr;
+        ir.lock();
+        try {
+            wr = injectors.get(cl);
+        } finally {
+            ir.unlock();
+        }
+        if (wr != null) {
+            injector = wr.get();
+        }
+        if (injector == null) {
+            try {
+                wr = new WeakReference<Injector>(injector = new Injector(cl));
+                iw.lock();
+                try {
+                    if (!injectors.containsKey(cl)) {
+                        injectors.put(cl, wr);
+                    }
+                } finally {
+                    iw.unlock();
+                }
+            } catch (SecurityException e) {
+                logger.log(Level.FINE, "Unable to set up a back-door for the injector", e);
+                return null;
+            }
+        }
+        return injector;
+    }
+    /**
+     * Injected classes keyed by their names.
+     */
+    private final Map<String, Class> classes = new HashMap<>();
+    private final ReentrantReadWriteLock rwl = new ReentrantReadWriteLock();
+    private final Lock r = rwl.readLock();
+    private final Lock w = rwl.writeLock();
+    private final ClassLoader parent;
+    /**
+     * True if this injector is capable of injecting accessors.
+     * False otherwise, which happens if this classloader can't see {@link Accessor}.
+     */
+    private final boolean loadable;
+    private static Method defineClass;
+    private static Method resolveClass;
+    private static Method findLoadedClass;
+    private static Object U;
+
+    static {
+        try {
+            Method[] m = AccessController.doPrivileged(
+                    new PrivilegedAction<Method[]>() {
+                @Override
+                public Method[] run() {
+                    return new Method[]{
+                        getMethod(ClassLoader.class, "defineClass", String.class, byte[].class, Integer.TYPE, Integer.TYPE),
+                        getMethod(ClassLoader.class, "resolveClass", Class.class),
+                        getMethod(ClassLoader.class, "findLoadedClass", String.class)
+                    };
+                }
+            }
+            );
+            defineClass = m[0];
+            resolveClass = m[1];
+            findLoadedClass = m[2];
+        } catch (Throwable t) {
+            try {
+                U = AccessController.doPrivileged(new PrivilegedExceptionAction() {
+                    @Override
+                    public Object run() throws Exception {
+                        Class u = Class.forName("sun.misc.Unsafe");
```

Comment:
this must be `jdk,internal.misc.Unsafe` instead otherwise the class won't be found on JDK11